proposals/lip-0049.md

```diff
@@ -487,25 +549,28 @@ sidechainTerminatedCCMParamsSchema = {
 * `chainID`: chain ID of the terminated chain,
 * `stateRoot`: last state root of the terminated sidechain, as certified on the mainchain.
 
-##### Creating Sidechain Terminated Message
+##### Creation
 
 A sidechain terminated message is created by the Interoperability module when a message should be routed to a terminated or inactive chain.
 
-##### Executing Sidechain Terminated Message
-
-When a sidechain terminated message `stm` is executed, the following is done:
+##### Execution
 
 ```python
-if stm.sendingChain == MAINCHAIN_ID:
-    if there exist an entry in the terminated state substore with storeKey = uint32be(stm.params.chainID):
-        return
-
-    createTerminatedStateAccount(stm.params.chainID, stm.params.stateRoot)
-else:
-    terminateChainInternal(stm.sendingChainID)
+def execute(ccm: CCM) -> None:
```

Comment:
Are you specifying two functions here at the same time, that should be implemented differently in the sidechain interoperability module and mainchain interoperability module? Otherwise, we could simplify the logic to:

```    
# Processing on the mainchain
if ownChainAccount.ID == MAINCHAIN_ID or ccm.sendingChainID != MAINCHAIN_ID:
    # This message should not be created on a sidechain.
    terminateChain(ccm.sendingChainID)
elif isLive(ccm.params.chainID):
    createTerminatedStateAccount(ccm.params.chainID, ccm.params.stateRoot)
```

Reply:
In general, I kept the same style in all these cases where I needed to define two different protocols, to make it as easily understandable as possible. In this specific case, I already updated it

Reply:
Okay, all good then.